hazelcast/src/main/java/com/hazelcast/config/NativeMemoryConfig.java

```diff
@@ -222,24 +224,62 @@ public NativeMemoryConfig setMetadataSpacePercentage(float metadataSpacePercenta
     }
 
     /**
-     * Returns the persistent memory directory (e.g. Intel Optane) to be used to store memory structures allocated by native
-     * memory manager.
-     * <p>
-     * Default value is {@code null}. It indicates that volatile RAM is being used.
-     * {@code null}
+     * Returns the persistent memory configuration this native memory
+     * configuration uses.
+     *
+     * @return the persistent memory configuration
      */
+    public PersistentMemoryConfig getPersistentMemoryConfig() {
+        return persistentMemoryConfig;
+    }
+
+    /**
+     * Sets the persistent memory configuration this native memory
+     * configuration uses.
+     *
+     * @param persistentMemoryConfig The persistent memory configuration to use
+     */
+    public void setPersistentMemoryConfig(PersistentMemoryConfig persistentMemoryConfig) {
+        this.persistentMemoryConfig = requireNonNull(persistentMemoryConfig);
+    }
+
+    /**
+     * Returns the persistent memory directory (e.g. Intel Optane) to be
+     * used to store memory structures allocated by native memory manager.
+     * If there are multiple persistent memory directories are defined in
+     * {@link #persistentMemoryConfig}, an {@link IllegalStateException}
+     * is thrown.
+     *
+     * @see PersistentMemoryConfig#getDirectoryConfigs()
+     * @deprecated Since 4.1 multiple persistent memory directories are
+     * supported. Please use {@link PersistentMemoryConfig#getDirectoryConfigs()}
+     * instead.
+     */
+    @Deprecated
     public String getPersistentMemoryDirectory() {
-        return persistentMemoryDirectory;
+        List<PersistentMemoryDirectoryConfig> directoryConfigs = persistentMemoryConfig.getDirectoryConfigs();
+        int directoriesDefined = directoryConfigs.size();
+        if (directoriesDefined > 1) {
+            throw new HazelcastException("There are multiple persistent memory directories configured. Please use "
+                    + "PersistentMemoryConfig.getDirectoryConfigs()!");
+        }
+
+        return directoriesDefined == 1 ? directoryConfigs.get(0).getDirectory() : null;
     }
 
     /**
-     * Sets the persistent memory directory (e.g. Intel Optane) to be used to store memory structures allocated by native memory
-     * manager.
+     * Sets the persistent memory directory (e.g. Intel Optane) to be used
+     * to store memory structures allocated by native memory manager. If
+     * the {@link #persistentMemoryConfig} already contains directory
+     * definition, it is overridden with the provided {@code directory}.
+     *
      * @param directory the persistent memory directory
      * @return this {@link NativeMemoryConfig} instance
+     * @see #getPersistentMemoryConfig()
+     * @see PersistentMemoryConfig#addDirectoryConfig(PersistentMemoryDirectoryConfig)
      */
     public NativeMemoryConfig setPersistentMemoryDirectory(String directory) {
```

Comment:
Should we should also add `@Deprecated` here?

Reply:
Agree, done.